### ISSUE TITLE
Windows enhancements

### ISSUE THREAD

**Issue by rjindael** [lrre-foss/rnr#5]
- embed icon in output binaries
- add runtime libraries to output binary path (or link statically?)

**Comment by rjindael** [comment]
for icon:
- we could use a rc file, but we'd need a separate ico file, unless we can automate this in CMake somehow?
  - for said rc file does this work on linux?
  - and if it doesn't, we'd need to use lists for headers/sources and append to it with the rc file if windows

cc @floralrainfall

**Comment by rjindael** [comment]
- Dynamic libraries now get copied to output directory as of aca8c1c
- Runtime libraries must be linked statically

**Comment by floralrainfall** [comment]
Adding a rc file that would only appear on Windows could be done as
```cmake
if (WIN32)
    enable_language("RC")
    set (WIN32_RESOURCES ${CMAKE_CURRENT_SOURCE_DIR}/resource.rc)
endif()
```

**Comment by rjindael** [comment]
Resources (such as the icon and product description) now embed in the executable as of b6fb19b, but DLL deployment still remains an issue. We have several options to explore here:
- Use [`windeployqt`](https://doc.qt.io/qt-6/windows-deployment.html) or some equivalent
- Use `GET_RUNTIME_RESOURCES` (a CMake utility that has its [caveats](https://stackoverflow.com/a/64009154).)
- Link all runtime libraries statically.

I removed windeployqt from the latest commit as it somehow stopped working; either way, it still didn't copy every necessary DLL since MinGW requires libstdc++/libgcc libraries. `-static-libgcc -static-libstdc++` returned no luck. Perhaps this is just my machine??? but IDK. :/

**Comment by rjindael** [comment]
FWIW, this is the last thing preventing me from enabling CI builds (#7) -- I'd like to get it working without any additional user tinkering

**Comment by rjindael** [comment]
for future reference: https://sourceforge.net/p/mingw-w64/support-requests/148/

> Use -static-libstdc++ / -static-libstdc++ in combination with -static so not only standard libraries are statically linked, but also any others (like -lwinpthread).

**Comment by rjindael** [comment]
the conclusion I've reached is that these libraries (libstdc++, libgcc) cannot be linked statically unless we use CMake's [install command](https://cmake.org/cmake/help/latest/command/install.html), as these libraries are likely thrown in by OGRE/Qt/some other package we use and we are thus unable to detect it. furthermore, there are other libraries (such as libwinpthread) that we can't statically link without a whole lot of work. so: I've decided to close this issue for now and focus on getting DLL deployment to work in CI (through a hack using ldd) and hope this work for now until we seek to make RNR more formal and up to spec with CMake.
